app/donate/page.tsx

```diff
@@ -0,0 +1,430 @@
+      message: "Consume By Timing is required."
+    }), 
+    foodImages: z.array(z.instanceof(File))
+  });
+
+  const nonCookedFormSchema = z.object({
+    foodName: z.string().min(2, {
+      message: "Food name must be at least 2 characters.",
+    }),
+    expiryDate: z.string().nonempty({
+      message: "Expiry Date is required.",
+    }),
+    foodCategory: z.string().nonempty({
+      message: "Food Category is required.",
+    }),
+    specialHandling: z.string(),
+    quantity: z.number().min(1, { 
+      message: "Quantity of food must be at least 1."
+    }),
+    foodImages: z.array(z.instanceof(File))
+  });
+
+  const cookedForm = useForm<z.infer<typeof cookedFormSchema>>({
+    resolver: zodResolver(cookedFormSchema),
+    defaultValues: {
+      foodName: "",
+      timePrepared: "",
+      consumeByTiming: "",
+      specialHandling: "",
+      numberOfServings: 0,
+      foodImages: []
+    },
+  });
+
+  const nonCookedForm = useForm<z.infer<typeof nonCookedFormSchema>>({
+    resolver: zodResolver(nonCookedFormSchema),
+    defaultValues: {
+      foodName: "",
+      expiryDate: "",
+      foodCategory: "",
+      specialHandling: "",
+      quantity: 0,
+      foodImages: []
+    },
+  });
+
+  function onSubmit(values: any) {
+    console.log(values);
+
+    // make api call to save donation details in mongodb 
+    if (foodType === "Cooked Food") { 
+      cookedForm.reset({
+        foodName: "",
+        timePrepared: "",
+        specialHandling: "", 
+        numberOfServings: 0
+      })
+    } else { 
+      nonCookedForm.reset({
+        foodName: "",
+        expiryDate: "",
+        foodCategory: "",
+        specialHandling: "",
+        quantity: 0
+      });
+      setSelectedCategory(""); 
+    }
+
+  }
+  const [selectedCategory, setSelectedCategory] = useState("");
+  const [foodType, setFoodType] = useState("Non-Cooked Food");
+  const [previewImages, setPreviewImages] = useState<string[]>([]);
+
+
+  const handleFileChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const files = Array.from(e.target.files || []);
+    if (files) {
+      const previews = files.map((file) => URL.createObjectURL(file));
+      setPreviewImages(previews);
+      // cookedForm.setValue("foodImages", files); // Store files in the form state
+    }
+  };
+
+
+  return (
+    <div className="bg-gray-100 min-h-screen p-8">
+      {/* Navigation Bar */}
+      <header className="flex justify-between items-center mb-8">
+        <div className="text-2xl font-bold">FoodToGive</div>
+        <nav className="space-x-4">
+          <a href="/" className="text-gray-700 hover:text-gray-900">Home</a>
+          <a href="/about" className="text-gray-700 hover:text-gray-900">About Us</a>
+          <a href="/donate" className="text-gray-700 hover:text-gray-900">Donate</a>
+          <a href="/account" className="text-gray-700 hover:text-gray-900">Account</a>
+        </nav>
+      </header>
```

Comment:
Once we merge the header component created by @KatyayaniPande, we can probably replace this part with the component for consistency.

Reply:
@calistatan03 The Header component has been merged into main and merged into this PR, if you wish, you could begin working on replacing this with the new Header component!